strawberryfields/ops.py

```diff
@@ -1140,6 +1140,11 @@ class S2gate(Gate):
     def __init__(self, r, phi=0.):
         super().__init__([r, phi])
 
+    def _apply(self, reg, backend, **kwargs):
+        p = self.p[0] * pf.exp(1j * self.p[1])
+        z = par_evaluate(p)
```

Comment:
We really should fix up the Strawberry Fields conventions, we are doing too many unnecessary conversions :laughing:

Reply:
Yes, this would be great, write every gate explicitly in terms of two real parameters.